src/wallet/api/schemas/walletApi.ts

```diff
@@ -0,0 +1,508 @@
+ */
+export const fullNodeTokenSchema = z.object({
+  uid: z.string(),
+  name: z.string(),
+  symbol: z.string(),
+  amount: bigIntCoercibleSchema,
+});
+
+/**
+ * Schema for full node transaction data.
+ * Contains all information about a transaction as seen by the full node.
+ */
+export const fullNodeTxSchema = z.object({
+  hash: z.string(),
+  nonce: z.string(),
+  timestamp: z.number(),
+  version: z.number(),
+  weight: z.number(),
+  parents: z.array(z.string()),
+  inputs: z.array(fullNodeInputSchema),
+  outputs: z.array(fullNodeOutputSchema),
+  tokens: z.array(fullNodeTokenSchema),
+  token_name: z.string().nullable(),
+  token_symbol: z.string().nullable(),
+  raw: z.string(),
+});
+
+/**
+ * Schema for full node transaction metadata.
+ * Contains additional information about a transaction's status and relationships.
+ */
+export const fullNodeMetaSchema = z.object({
+  hash: z.string(),
+  received_by: z.array(z.string()),
+  children: z.array(z.string()),
+  conflict_with: z.array(z.string()),
+  first_block: z.string().nullable(),
+  height: z.number(),
+  voided_by: z.array(z.string()),
+  spent_outputs: z.array(z.tuple([z.number(), z.array(z.string())])),
+  received_timestamp: z.number().nullable(),
+  is_voided: z.boolean(),
+  verification_status: z.string(),
+  twins: z.array(z.string()),
+  accumulated_weight: z.number(),
+  score: z.number(),
+});
+
+/**
+ * Response schema for full node transaction data.
+ * Contains the transaction details, metadata, and optional message.
+ */
+export const fullNodeTxResponseSchema = baseResponseSchema.extend({
+  tx: fullNodeTxSchema,
+  meta: fullNodeMetaSchema,
+  message: z.string().optional(),
+  spent_outputs: z.record(z.string()).optional(),
+});
+
+/**
+ * Response schema for transaction confirmation data.
+ * Contains information about the transaction's confirmation status and weight.
+ */
+export const fullNodeTxConfirmationDataResponseSchema = baseResponseSchema.extend({
+  accumulated_weight: z.number(),
+  accumulated_bigger: z.boolean(),
+  stop_value: z.number(),
+  confirmation_level: z.number(),
+});
+
+/**
+ * Response schema for wallet status.
+ * Contains information about the wallet's current state.
+ */
+export const walletStatusResponseSchema = baseResponseSchema.extend({
+  status: z.object({
+    walletId: z.string(),
+    xpubkey: z.string(),
+    status: z.string(),
+    maxGap: z.number(),
+    createdAt: z.number(),
+    readyAt: z.number().nullable(),
+  }),
+  error: z.string().optional(),
+});
+
+/**
+ * Response schema for token list.
+ * Contains an array of token information.
+ */
+export const tokensResponseSchema = baseResponseSchema.extend({
+  tokens: z.array(z.string()),
+});
+
+/**
+ * Response schema for transaction history.
+ * Contains an array of transaction information.
+ */
+export const historyResponseSchema = baseResponseSchema.extend({
+  history: z.array(
+    z.object({
+      txId: z.string(),
```

Comment:
suggestion(non-blocking): we have many places where we use a tx hash. We could create a regex schema for it, instead of just string.

Reply:
Done!